### PR TITLE
Fix query failures due to incorrect partition casting

### PR DESCRIPTION
**Pull request by matteofigus**
*Description of changes*
Adds handling for non-string partition types such as int. I opted for keeping the strategy consistent with what we do with matches: we get the partition type from glue, then we cast it during the generate phase, so that the execute inherits the values already casted. 

Credits: I took some code snippets from @ormahler's #188 PR 

*PR Checklist:*

- [x] Changelog updated
- [x] Unit tests (and integration tests if applicable) provided
- [x] All tests pass
- [x] Pre-commit checks pass
- [x] Debugging code removed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/tasks/test_generate_queries.py [hunk 1]
```diff
@@ -622,9 +662,17 @@ def test_it_throws_for_unconvertable_matches(self):
...
+        with pytest.raises(ValueError) as e:
             get_inner_children("struct<name:string", "struct<", ">")
+        assert e.value.args[0] == "Column schema is not valid"
+
+    def test_it_throws_for_invalid_schema_for_nested_children(self):
+        with pytest.raises(ValueError) as e:
+            get_nested_children(
+                "struct<name:string,age:int,s:struct<n:int>,b:string", "struct"
+            )
+        assert e.value.args[0] == "Column schema is not valid"
```
Comment (matteofigus) [inline tests/unit/tasks/test_generate_queries.py:675]: This is slightly unrelated, there was no coverage for it so I just added those to make sure that scenario is handled

File: tests/unit/tasks/test_execute_query.py [hunk 2]
```diff
@@ -68,6 +68,42 @@ def test_it_generates_query_with_partition():
...
+            "PartitionKeys": [{"Key": "product_category", "Value": "Books"}],
+        }
+    )
+
+    assert (
+        escape_resp(resp) == 'SELECT DISTINCT "$path" '
+        'FROM "amazonreviews"."amazon_reviews_parquet" '
+        'WHERE ("customer_id" in (123456, 456789)) '
+        "AND \"product_category\" = 'Books'"
+    )
```
Comment (matteofigus) [inline tests/unit/tasks/test_execute_query.py:86]: Slightly unrelated but there was no test on the execute query for int matches so I added it just for clarity

File: tests/unit/tasks/test_generate_queries.py [hunk 3]
```diff
@@ -690,7 +738,9 @@ def partition_stub(values, columns, table_name="test_table"):
     }
 
 
-def table_stub(columns, partition_keys, table_name="test_table"):
+def table_stub(
+    columns, partition_keys, table_name="test_table", partition_key_types="string"
```
Comment (HieronymusLex) [inline tests/unit/tasks/test_generate_queries.py:742]: nit: types? shouldn't this be type if the value isn't a list?

File: tests/acceptance/conftest.py [hunk 4]
```diff
@@ -256,6 +256,7 @@ def factory(
         prefix="prefix",
         partition_keys=[],
         partitions=[],
+        partition_key_types="string",
```
Comment (HieronymusLex) [inline tests/acceptance/conftest.py:259]: nit: types? shouldn't this be type if the value isn't a list?

**Review by HieronymusLex — APPROVED** [review]
couple of very minor comments. LGTM though

File: tests/unit/tasks/test_generate_queries.py (hunk 3 above)
Comment (matteofigus) [reply tests/unit/tasks/test_generate_queries.py]: you're right, renamed to partition_keys_type

**Review by HieronymusLex — APPROVED** [review]

**Pull request merged**